### PR TITLE
COMMERCE-8885 info_box refactoring

### PR DESCRIPTION
**Pull request by garofalog**

### CODE REVIEW

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/info_box/page.jsp [hunk 1]
```diff
@@ -27,27 +27,26 @@ String linkId = PortalUtil.generateRandomKey(request, "info-box") + "_action-lin
...
 		</c:if>
 	</header>
+</div>
 
-	<div class="description">
\ No newline at end of file
+<liferay-frontend:component
+	context='<%=
+		HashMapBuilder.<String, Object>put(
+			"actionLabelValidator", Validator.isNotNull(actionLabel)
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/info_box/page.jsp:42]: You can remove it and validate the actionLabel directly into the component.

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/info_box/page.jsp [hunk 2]
```diff
@@ -27,27 +27,26 @@ String linkId = PortalUtil.generateRandomKey(request, "info-box") + "_action-lin
...
+		).put(
+			"actionTargetId", HtmlUtil.escapeJS(actionTargetId)
+		).put(
+			"actionTargetIdValidator", Validator.isNotNull(actionTargetId)
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/info_box/page.jsp:46]: Same here, you can validate it in the component.

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/js/info_box/page.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	const link = document.getElementById(linkId);
+
+	function openModal(event) {
+		event.preventDefault();
+		Liferay.fire(OPEN_MODAL, {
+			id: actionTargetId,
+		});
+	}
+
+	if (link && actionLabelValidator && actionTargetIdValidator) {
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/js/info_box/page.js:18]: Create a const to store the result of this if statement.

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@garofalog Please take a look at the comments, thanks!

**Comment by giuliogarofalo** [comment]
resolved in [this](https://github.com/liferay-commerce/liferay-portal/pull/2592) pr. this will be closed

**Comment by garofalog** [comment]
resolved in [this](https://github.com/liferay-commerce/liferay-portal/pull/2592) pr. this pr will be closed then.

**Pull request closed without merging**